File: DataLayer/Entities/Player.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace DataLayer.Entities;

public class Player
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int? Id { get; set; }

    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Alias { get; set; }
    public int Age { get; set; }
    public string Gender { get; set; }
    public string Nationality { get; set; }
    public int Overall { get; set; }
    public int GameSense { get; set; }
    public int TeamFighting { get; set; }
    public int Dueling { get; set; }
    public int JglPathing { get; set; }
    public int WaveMgmt { get; set; }
    public int Farming { get; set; }
    public List<Team> Teams { get; set; }
    public List<Member> Members { get; set; }
    public List<Series> Series { get; set; }
}

File: frontend/src/BusinessLogic/Models/Player.ts
import ApiReturn from "./ApiReturn";

export default class Player extends ApiReturn {
    url: string;
    firstName: string;
    lastName: string;
    alias: string;
    age: number;
    gender: string;
    nationality: string;
    overall: number;
    gameSense: number;
    teamFighting: number;
    dueling: number;
    jglPathing: number;
    waveMgmt: number;
    farming: number;
    currentTeamUrl: string;
    currentTeam: string;

    constructor(data = null) {
        super();
        if (data == null) return;
        this.url = data.url;
        this.firstName = data.firstName;
        this.lastName = data.lastName;
        this.alias = data.alias;
        this.age = data.age;
        this.gender = data.gender;
        this.nationality = data.nationality;
        this.overall = data.overall;
        this.gameSense = data.gameSense;
        this.teamFighting = data.teamFighting;
        this.dueling = data.dueling;
        this.jglPathing = data.jglPathing;
        this.waveMgmt = data.waveMgmt;
        this.farming = data.farming;
        this.currentTeamUrl = data.currentTeamUrl;
        this.currentTeam = data.currentTeam;

        this.loading = false;
    }
}